beacon_node/beacon_chain/src/kzg_utils.rs

```diff
@@ -147,15 +147,16 @@ mod test {
         type E = MainnetEthSpec;
         let num_of_blobs = 6;
         let spec = E::default_spec();
-        let (signed_block, blob_sidecars) = create_test_block_and_blobs::<E>(num_of_blobs, &spec);
+        let (signed_block, blobs) = create_test_block_and_blobs::<E>(num_of_blobs, &spec);
 
         let trusted_setup: TrustedSetup = serde_json::from_reader(TRUSTED_SETUP_BYTES)
             .map_err(|e| format!("Unable to read trusted setup file: {}", e))
             .expect("should have trusted setup");
         let kzg = Kzg::new_from_trusted_setup(trusted_setup).expect("should create kzg");
 
+        let blob_refs = blobs.iter().collect::<Vec<_>>();
```

Comment:
I guess it might make sense for kzg libraries to take in a `impl Iterator<Item = Blob>` to possibly avoid these allocations

Reply:
@michaelsproul and i made an attempt today but it touches too much existing code, we might refactor this in a separate PR.